### PR TITLE
nap: init at 0.1.1

### PR DESCRIPTION
**Pull request by leolavaur**
Add initial version of maaslalani/nap, a code snippet manager.

###### Description of changes

I added [nap](https://github.com/maaslalani/nap), using last release (v0.1.1). `nap` provides an interface to manage code snippets from a terminal. It is currently maintained by @maaslalani, but I took the liberty to make this first update.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/nap/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  vendorSha256 = "sha256-puCqql77kvdWTcwp8z6LExBt/HbNRNe0f+wtM0kLoWM=";
+
+  excludedPackages = ".nap";
+
+  meta = {
+    description = "Code snippets in your terminal 🛌";
+    homepage = "https://github.com/maaslalani/nap";
+    license = lib.licenses.mit;
+    maintainers = [ "phdcybersec" ];
```
Comment (maaslalani) [inline pkgs/development/tools/nap/default.nix:22]: Feel free to add me as a maintainer here if you'd like :)

**Review by maaslalani — APPROVED** [review]
Thanks so much @phdcybersec!

File: pkgs/development/tools/nap/default.nix (hunk 1 above)
Comment (phdcybersec) [reply pkgs/development/tools/nap/default.nix]: Done! Thanks :)

File: pkgs/development/tools/nap/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  vendorSha256 = "sha256-puCqql77kvdWTcwp8z6LExBt/HbNRNe0f+wtM0kLoWM=";
+
+  excludedPackages = ".nap";
+
+  meta = {
+    description = "Code snippets in your terminal 🛌";
+    homepage = "https://github.com/maaslalani/nap";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ "phdcybersec" "maaslalani" ];
```
Comment (raphaelr) [inline pkgs/development/tools/nap/default.nix:22]:
```suggestion
    maintainers = with lib.maintainers; [ phdcybersec maaslalani ];
```
Need to remove the quotes

**Review by raphaelr — COMMENTED** [review]
Need to remove the quotes in meta.maintainers

Then squash all commits into one with message `nap: init at 0.1.1`

File: pkgs/development/tools/nap/default.nix (hunk 2 above)
Comment (phdcybersec) [reply pkgs/development/tools/nap/default.nix:22]: Alright will do. Sorry, this my first experience packaging software for Nix.

Comment (raphaelr) [reply pkgs/development/tools/nap/default.nix]: > Alright will do. Sorry, this my first experience packaging software for Nix.
No worries!

**Review by raphaelr — APPROVED** [review]
Result of `nixpkgs-review pr 202261` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nap</li>
  </ul>
</details>

Tested basic functionality; works

**Review by BenediktBroich — APPROVED** [review]

**Review by maaslalani — APPROVED** [review]

**Pull request merged**